### PR TITLE
Feature/aws variant

### PR DESCRIPTION
**Pull request by lleadbet**
This PR is one of 4 PRs to bring AWS Support to BAS. This PR should likely be merged last of those, as it adds new secrets to the existing BAS Subgraph/Infra repos, as well as not generating GCP credentials (for hopefully obvious reasons). 

Of note: 

* This moves all Terraform into a new root-level `terraform` folder, designed to hold any cloud providers. Right now, this is just GCP and AWS. 
* Uses existing commands, so once you get into a K8S cluster, the commands should be identical (or nearly- a few small places where it differs)
* Updates docs to include any specific AWS callouts.

### CODE REVIEW

**Comment by lennyburdette** [comment]
looks like you might need to remove some commits from your rebase. doesn't look like the diagrams changed but they're included in the diff for some reason

File: 04-observability/README.md [hunk 1]
```diff
@@ -18,15 +18,29 @@ The router and subgraphs are already configured to send Open Telemetry traces to
...
-Make a GraphQL request to the router via its IP address:
+Make a GraphQL request to the router via its IP address. 
+
+### GCP
 
 ```sh
 kubectx apollo-supergraph-k8s-prod
 ROUTER_IP=$(kubectl get ingress -n router -o jsonpath="{.*.*.status.loadBalancer.ingress.*.ip}")
 open http://$ROUTER_IP
+kubectl port-forward -n zipkin svc/zipkin 9411:9411
```
Comment (lennyburdette) [inline 04-observability/README.md:29]: if you break this into two separate code blocks, they'll each over their own "copy" button

Comment (lennyburdette) [reply 04-observability/README.md:29]: and you can't pasted both the `kubectl port-forward` line followed by the `open` line. i switched to just having one code block for port-forward and then a markdown link

File: terraform/aws/github.tf [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+resource "github_actions_secret" "subgraph_aws_secret" {
+  for_each = {
+    for index, subgraph in var.subgraphs : subgraph.name => subgraph
+  }
+  repository      = github_repository.subgraph_repos[each.key].name
+  secret_name     = "AWS_SECRET_KEY"
+  plaintext_value = aws_iam_access_key.eks_user_key.secret
+}
+
+# infra repo secrets: the only different value is GCP_CREDENTIALS
```
Comment (lennyburdette) [inline terraform/aws/github.tf:83]: are you using the same secret for all three repos? this line is incorrect regardless

File: 04-observability/README.md (hunk 1 above)
Comment (lleadbet) [reply 04-observability/README.md:29]: Weird, that command had worked for me when testing- can split out like above, however.

File: terraform/aws/github.tf (hunk 2 above)
Comment (lleadbet) [reply terraform/aws/github.tf:83]: Yeah, I can update the comment but the secret is the same for all three. The token itself is actually pretty much useless for the AWS console, just has access to the cluster itself.

**Comment by lleadbet** [comment]
@lennyburdette okay i think i finally sorted this silly rebase issue.

File: 01-setup/README.md [hunk 3]
```diff
@@ -88,12 +125,19 @@ gcloud services enable \
...
 
-```
+#### If using AWS
+
+```sh
+aws configure
 gh auth login
 ```
 
+#### General
```
Comment (lennyburdette) [inline 01-setup/README.md:138]: i can't leave a comment on line 167, but you might want to grep for `cd 01-setup` to change it to `cd terraform/{gcp/aws}` or something

File: 01-setup/README.md [hunk 4]
```diff
@@ -64,7 +98,7 @@ export TF_VAR_github_token="<your github personal access token>"
 Run this script to create your graph and get environment variables for GraphOS:
 
 ```sh
-cd 01-setup
+cd terraform/gcp
```
Comment (lennyburdette) [inline 01-setup/README.md:101]: i'm wondering if we should change these to 
```
# in terraform/aws or terraform/gcp
source .env
./create_graph.sh
```

File: 02-managed-federation/README.md [hunk 5]
```diff
@@ -56,30 +62,44 @@ In both **subgraph-a** and **subgraph-b** repositories:
...
+### <image src="../images/gcp.svg" height="13" style="margin:auto;" /> GCP
 
 ```sh
+kubectx apollo-supergraph-k8s-prod
 ROUTER_IP=$(kubectl get ingress -n router -o jsonpath="{.*.*.status.loadBalancer.ingress.*.ip}")
 open http://$ROUTER_IP
 ```
 
-The Google Cloud ingress may take a few minutes to start. If you don't want to wait for an IP address you can use `port-forward`:
+Lastly, open up [http://localhost:9411/](http://localhost:9411/) in addition to the Router page. Upon doing so, you'll have both the Router and Zipkin pages open. Run a few requests and refresh the Zipkin list, and you should be seeing them come through!
```
Comment (lennyburdette) [inline 02-managed-federation/README.md:86]: why are you covering zipkin in 02-managed-federation?

Comment (lleadbet) [reply 02-managed-federation/README.md:86]: 🤦 that's what I get for copy/pasting. Will fix.

**Review by lennyburdette — APPROVED** [review]

**Pull request merged**